### DISCUSSION TITLE
How to get window records in sink

### DISCUSSION THREAD

**Discussion by khan-lau** [lf-edge/ekuiper#2936] [Use Case]
我需要在sink中将rule聚合的结果再做一下处理然后入库. 
处理方式例如补偿,  补记录等等. 此时需要获取窗口原始记录

**Comment by ngjaying** [comment]
I don't understand your problem. Could you please describe with data, not just natural language. What's the expected data? What did you see now?

**Comment by khan-lau** [comment]
example data :
| index | name | value  | 
| ------ | ------ | ------- |
| 1        | a        | 1.0      |
| 2        | a        | 2.0      |
| 3        | a        | 3.0      |
| 4        | a        | 100.0  |
| 5        | a        | 100.0  |
| 6        | a        | 6.0      |

COUNTWINDOW(2, 1)

record 4 的 value 与 record 3 的`差值`太大, 我称之为`跳变`, 需要根据一定规则将其修改为4.0, 
record 5 的 value 与修改后的 record 4 的`差值`也太大, 需要修改为5.0

record 1 2 3 6 应该是正常值, 

由于我无法修改window中的记录, 
所以 :
record 3-4 `跳变`, `不符合`预期,  
record 4-5 本应该也是`跳变`, `不符合`预期, 
record 5-6 本应该是`恢复正常`, 现在成了跳变, `不符合`预期

表达能力有限, 还请谅解, 不知道我解释清楚了没有

**Reply by ngjaying** [reply]
规则 SendSingle=false， sink可以拿到整个窗口的数据。本身这个跳变需求可能也不需要写插件，用分析函数可以拿到前值跟当前的值做对比然后做修正，参考 https://ekuiper.org/docs/en/latest/example/change_data_capture.html 和 https://ekuiper.org/docs/en/latest/sqls/functions/analytic_functions.html

**Reply by khan-lau** [reply]
我需要拿修改后的前值跟当前值对比,  单纯的拿前值在碰到连续跳变的时候会产生干扰
